internal/test/e2e/cluster_test.go

```diff
@@ -595,25 +595,13 @@ func TestUploadDownloadBasic(t *testing.T) {
 	contracts, err := cluster.Bus.Contracts(context.Background(), api.ContractsOpts{})
 	tt.OK(err)
 
-	// assert contracts haven't been revised
+	// collect the current revision heights
+	revisionHeights := make(map[types.FileContractID]uint64)
 	for _, c := range contracts {
-		if c.RevisionHeight != 0 {
```

Comment:
Why is the revision height not 0 before mining the contract?

Reply:
In `processContract` I now always pass a revision number and file size, regardless of whether the contract was actually revised because we now no longer consider a formation + revision in the same block a real revision. I wasn't too sure if I could do that, I considered adding an extra `revised` bool there to make the distinction still and always update the file size but only update the revision height and number if it was actually revised.

To make sure the test is closer to how it was originally intended I moved it into `TestContractApplyChainUpdates`, where I create a cluster without autopilot and form a contract manually. That way I can assert the revision height is indeed 0 because the contract hasn't been mined yet and nothing is interfering with it. After broadcasting its revision and mining another block I can assert the revision height got updated.